src/main/java/org/folio/circulation/support/CollectionResourceClient.java

```diff
@@ -185,37 +158,46 @@ private static boolean isProvided(String query) {
    * @param pageOffset  the value for the offset parameter, may be null for none
    * @return the query string, may be empty
    */
-  static String createQueryString(
-    String urlEncodedCqlQuery,
-    Integer pageLimit,
+  static String createQueryString(String urlEncodedCqlQuery, Integer pageLimit,
     Integer pageOffset) {
 
-    String query = "";
+    String queryParameter = prefixOnCondition(
+      "query=", urlEncodedCqlQuery, CollectionResourceClient::isProvided);
+
+    String limitParameter = prefixOnCondition(
+      "limit=", pageLimit, Objects::nonNull);
+
+    String offsetParameter = prefixOnCondition(
+      "offset=", pageOffset, Objects::nonNull);
+
+    final String queryStringParameters
+      = Stream.of(queryParameter, limitParameter, offsetParameter)
+      .filter(StringUtils::isNotBlank)
+      .collect(Collectors.joining("&"));
+
+    return prefixOnCondition(
+      "?", queryStringParameters, StringUtils::isNotBlank);
+  }
+
+  private static <T> String prefixOnCondition(String prefix, T value,
```

Comment:
Maybe something like, or similar?
```suggestion
  private static <T> String concatenateIf(String prefix, T value,
```

Reply:
Good idea. That would be a better name, thank you. 

Would it be ok if I don't apply that name change, because all of that code is removed in a subsequent pull requests, replaced with typed query parameters and (hopefully) better combination logic.

Reply:
Yes, that's fine. That was another idea to use an object instead.

Thanks.